src/test/java/codeu/controller/AdminServletTest.java

```diff
@@ -0,0 +1,69 @@
+
+
+package codeu.controller;
+
+import static org.junit.Assert.fail;
+
+import java.io.IOException;
+
+import javax.servlet.RequestDispatcher;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class AdminServletTest {
+
+    private AdminServlet adminServlet;
+    private HttpServletRequest mockRequest;
+    private HttpSession mockSession;
+    private HttpServletResponse mockResponse;
+    private RequestDispatcher mockRequestDispatcher;
+
+
+    @Before
+    public void setup() {
+        adminServlet = new AdminServlet();
+        try {
+			adminServlet.init();
```

Comment:
The pattern the rest of the servlet classes follow is to not call the full init() method from the unit tests, because it might include too much heavy-weight setup (like if we want to initialize a database connection later).

Instead, you could extract a protected `setAdminUsernames` method, and call that from both the main init method and also this test setup. Does that make sense?

Reply:
Will do!